android/modules/media/src/java/ti/modules/titanium/media/TiCameraXActivity.java

```diff
@@ -0,0 +1,629 @@
+		}
+	}
+
+	@SuppressLint("MissingPermission")
+	public static void startVideoCapture()
+	{
+		KrollDict recordingDict = new KrollDict();
+
+		Consumer<VideoRecordEvent> vre = videoRecordEvent -> {
+			if (videoRecordEvent instanceof VideoRecordEvent.Start) {
+				// Handle the start of a new active recording
+			} else if (videoRecordEvent instanceof VideoRecordEvent.Pause) {
+				// Handle the case where the active recording is paused
+			} else if (videoRecordEvent instanceof VideoRecordEvent.Resume) {
+				// Handles the case where the active recording is resumed
+			} else if (videoRecordEvent instanceof VideoRecordEvent.Finalize) {
+				VideoRecordEvent.Finalize finalizeEvent =
+					(VideoRecordEvent.Finalize) videoRecordEvent;
+				// Handles a finalize event for the active recording, checking Finalize.getError()
+				int error = finalizeEvent.getError();
+				if (error != VideoRecordEvent.Finalize.ERROR_NONE) {
+					// error
+				}
+				Uri uri = finalizeEvent.getOutputResults().getOutputUri();
+
+				if (uri.toString().startsWith("content://")) {
+					String[] columns = {
+						MediaStore.Video.VideoColumns.TITLE,
+						MediaStore.Video.VideoColumns.DATA
+					};
+					String[] selectionArgs = { mediaTitle };
+					createFile(
+						MediaStore.Video.Media.EXTERNAL_CONTENT_URI,
+						columns,
+						MediaStore.Video.VideoColumns.TITLE + " = ? ",
+						selectionArgs
+					);
+				} else {
+					// normal file
+					File file = new File(uri.getPath());
+					TiBlob blob = TiBlob.blobFromFile(TiFileFactory.createTitaniumFile(file.getPath(), false));
+					KrollDict response = MediaModule.createDictForImage(blob, blob.getMimeType());
+					successCallback.callAsync(callbackContext, response);
+				}
+
+				if (cameraActivity != null && autohide) {
+					if (recording != null) {
+						recording.close();
+						recording = null;
+					}
+					cameraActivity.finish();
+				}
+			}
+			// All events, including VideoRecordEvent.Status, contain RecordingStats.
+			// This can be used to update the UI or track the recording duration.
+			RecordingStats recordingStats = videoRecordEvent.getRecordingStats();
+			long recordingTime = (recordingStats.getRecordedDurationNanos() / 1000 / 1000);
+			if (isRecording && videoMaximumDuration > 0 && recordingTime >= videoMaximumDuration) {
+				stopVideoCapture();
+			}
+			if (isRecording && recordingCallback != null) {
+				recordingDict.put("duration", recordingTime);
+				recordingDict.put("size", recordingStats.getNumBytesRecorded());
+				recordingCallback.callAsync(callbackContext, recordingDict);
+			}
+		};
+
+		if (cameraActivity.hasAudioRecorderPermissions()) {
+			recording = pendingRecording.withAudioEnabled().start(executor, vre);
+			isRecording = true;
+		} else {
+			recording = pendingRecording.start(executor, vre);
+			isRecording = true;
+		}
+	}
+
+	public static void pauseVideoCapture()
+	{
+		if (recording != null) recording.pause();
```

Comment:
Are inline if-conditions recommended in our style guide? In JS, it's mostly disallowed these days